src/components/Header/Header.jsx

```diff
@@ -0,0 +1,128 @@
+import { Link } from "react-router-dom";
+
+const Header = () => {
+  const [scrollProgress, setScrollProgress] = useState(0);
+  const [isMenuOpen, setIsMenuOpen] = useState(false); // To control the mobile slider
+
+  // Detect if the device is mobile using a media query hook
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const scrollPosition = window.scrollY;
+      const maxScroll = 10;
+      const progress =
+        scrollPosition > maxScroll ? 1 : scrollPosition / maxScroll;
+      setScrollProgress(progress);
+    };
+
+    window.addEventListener("scroll", handleScroll);
+    return () => window.removeEventListener("scroll", handleScroll);
+  }, []);
+
+  const headerStyle = {
+    width: `${100 - (1 - scrollProgress) * 20}%`,
+    maxWidth: "1536px",
+    transform: `scale(${0.9 + scrollProgress * 0.1})`,
+    borderRadius: `${(1 - scrollProgress) * 0.75}rem`,
+  };
+
+  // Toggle menu open/close
+  const toggleMenu = () => setIsMenuOpen((prev) => !prev);
+
+  // Mobile slider navigation
+  const renderMobileNav = () => (
+    <div
+      className={`fixed top-0 left-0 h-full w-64 bg-headerBackground z-50 transition-transform duration-300 ease-in-out transform ${
+        isMenuOpen ? "translate-x-0" : "-translate-x-full"
+      }`}
+    >
+      <div className="flex justify-between p-4 items-center border-b border-gray-200">
+        <h1 className="text-[20px] font-bold text-headerBrand">Menu</h1>
+        <button onClick={toggleMenu} className="text-gray-500 text-2xl">
+          <FaTimes />
+        </button>
+      </div>
+      <div className="flex flex-col p-4 space-y-4">
+        <Link to="/" onClick={toggleMenu} className="text-[16px] font-semibold">
+          Home
+        </Link>
+        <Link
+          to="https://github.com/utk09-NCL/color-palette-generator/"
+          onClick={toggleMenu}
+          className="text-[16px] font-semibold"
+        >
+          GitHub
+        </Link>
+      </div>
+    </div>
+  );
+
+  return (
+    <div
+      className={`${
+        scrollProgress > 0 ? "pt-0" : "pt-4"
+      } fixed top-0 z-[9999] w-full flex justify-center items-center transition-all duration-300 ease-in-out`}
+    >
+      <>
+        {/* Mobile View */}
+        <header className="bg-headerBackground fixed top-0 w-full z-50 p-4 lg:hidden">
+          <div className="relative flex items-center justify-between w-full">
+            {/* Hamburger Icon */}
+            <button
+              onClick={toggleMenu}
+              className="text-gray-500 text-2xl focus:outline-none"
+            >
+              <FaBars />
+            </button>
+            {/* Logo */}
+            <h1 className="text-[20px] font-extrabold text-headerBrand whitespace-nowrap">
+              Color Conjure
+            </h1>
+            {/* GitHub Link */}
+            <Link
+              to={"https://github.com/utk09-NCL/color-palette-generator/"}
+              className="w-6 h-6"
+            >
```

Comment:
Please make this a 
`<a href="https://github.com/utk09-NCL/color-palette-generator" target="_blank" rel="noreferrer">...</a>`

Reply:
okay